app/controllers/sales_controller.rb

```diff
@@ -0,0 +1,25 @@
+class SalesController < ApplicationController
+  def new
+    @sale = Sale.new
+  end
+
+  def create
+    @sale = Sale.new(sale_create_params[:sale])
+    if @sale.save
+      redirect_to vendor_path(@sale.vendor_id)
+    else
+    render :new
+    end
+  end
+
+
+  private
+  #tells us what parameters we want to use when we create an album
+  def sale_create_params
+    params.permit(sale: [:amount, :vendor_id, :market_id, :purchase_time, :product_id])
```

Comment:
It's just a matter of style, but I feel like the `.require` way of writing this is clearer, specially when interacting with the code later within controller methods -- you wouldn't have to reference the key `:sale` within the `sale_create_params` hash. You did a great job with this in the markets controller!